src/error.rs

```diff
@@ -85,6 +88,8 @@ impl<'tcx> fmt::Display for EvalError<'tcx> {
                 write!(f, "array index {} out of bounds {} at {:?}", index, len, span),
```

Comment:
The typical message is `index out of bounds: the len is {len} but the index is {index}`.